main.go

```diff
@@ -0,0 +1,332 @@
+package main
+
+import (
+	"encoding/json"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/Diminho/MK_practice/config"
+	"github.com/Diminho/MK_practice/models"
+	"github.com/Diminho/MK_practice/simplelog"
+	logjson "github.com/Diminho/MK_practice/simplelog/handlers/json"
+	"github.com/gorilla/websocket"
+	"golang.org/x/oauth2"
+)
+
+type App struct {
+	broadcast      chan models.EventPlaces
+	eventClients   map[string][]*websocket.Conn
+	facebookState  string
+	facebookConfig *oauth2.Config
+	slog           *simplelog.Log
+	db             func() *models.DB
+}
+
+type EventSystemMessage struct {
+	Message        string `json:"sysMessage"`
+	MessageType    int    `json:"messageType"`
+	Event          string `json:"event"`
+	LastActedPlace string `json:"lastActedPlace"`
+	BookTime       int    `json:"bookTime"`
+}
+
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+var mu = &sync.Mutex{}
+
+func main() {
+
+	file, fileErr := os.OpenFile("log.txt", os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if fileErr != nil {
+		log.Fatal("Failed to open log file: ", fileErr)
+	}
+	defer file.Close()
+
+	slog := simplelog.NewLog(file)
+	slog.SetHandler(logjson.New(slog))
+
+	db, dbErr := models.Connect()
+	if dbErr != nil {
+		slog.Fatal(dbErr)
+	}
+
+	app := &App{
+		eventClients: make(map[string][]*websocket.Conn),
+		broadcast:    make(chan models.EventPlaces),
+		facebookConfig: &oauth2.Config{
+			ClientID:     config.ClientID,
+			ClientSecret: config.ClientSecret,
+			RedirectURL:  config.RedirectURL,
+			Scopes:       config.Scopes,
+			Endpoint:     config.Endpoint,
+		},
+		facebookState: "MK_PRACTICE",
+		db:            db.DBInstance(),
+		slog:          slog,
+	}
+
+	fs := http.FileServer(http.Dir("./public"))
+	http.Handle("/", fs)
+
+	http.HandleFunc("/ws", app.handleConnections)
+	http.HandleFunc("/event", app.handleEvent)
+	http.HandleFunc("/facebook_login", app.handleFacebookLogin)
+	http.HandleFunc("/facebookCallback", app.handleFacebookCallback)
+	go app.handlePlaceBookings()
+
+	//simple graceful shutdown
+	var gracefulShut = make(chan os.Signal)
+	signal.Notify(gracefulShut, syscall.SIGTERM)
+	signal.Notify(gracefulShut, syscall.SIGINT)
+	go func() {
+		sig := <-gracefulShut
+		slog.Infof("caught signal: %+v", sig)
+		slog.Info("Wait for 5 second to finish processing")
+		time.Sleep(5 * time.Second)
+		os.Exit(0)
+	}()
+
+	slog.Info("Server running on port :8000")
+	err := http.ListenAndServe(":8000", nil)
```

Comment:
how incoming connections stops processing in case graceful shutdown was initiated?